### ISSUE TITLE
Add generic settings for metadata

### ISSUE THREAD

**Issue by guerlab** [spring-cloud/spring-cloud-commons#1084]
**Is your feature request related to a problem? Please describe.**
At present, spring cloud does not support the general setting of metadata and depends on the internal implementation of service registration and discovery client. When switching the implementation, the configuration file needs to be reconfigured

**Describe the solution you'd like**
Add generic configuration properties (spring.cloud.discovery.metadata) to support generic processing

**Describe alternatives you've considered**

**Additional context**

Configuration file before implementation
```
spring:
  cloud:
    consul:
      discovery:
        metadata: ## consul metadata configs
    nacos:
      discovery:
        metadata: ## nacos metadata configs
eureka:
  instance:
    metadata-map: ## eureka metadata configs

```

Configuration file after implementation
```
spring:
  cloud:
    discovery:
      metadata:
```

**Comment by huifer** [comment]
Based on [issues#999](https://github.com/spring-cloud/spring-cloud-commons/issues/999) part of metadata related content is designed，I don't know if that fits your idea.

**Comment by guerlab** [comment]
Meta information processing before service registration can be realized by listening to org.springframework.cloud.client.discovery.event.InstancePreRegisteredEvent.class

```
	@EventListener(org.springframework.cloud.client.discovery.event.InstancePreRegisteredEvent.class)
	public void instancePreRegistered(org.springframework.cloud.client.discovery.event.InstancePreRegisteredEvent event) {
		Map<String, String> metadata = event.getRegistration().getMetadata();

		metadata.put("foo", "xoo");
	}
```

**Comment by guerlab** [comment]
Implemented in 2022.0.0-m3
